corehq/blobs/s3db.py

```diff
@@ -100,7 +100,10 @@ def get_path(self, identifier=None, bucket=DEFAULT_BUCKET):
 
 
 def safepath(path):
-    if path.startswith(("/", ".")) or ".." in path or not SAFENAME.match(path):
+    if (path.startswith(("/", ".")) or
+            "/../" in path or
+            path.endswith("/..") or
```

Comment:
i don't have too much context but do you also need to check if `path.startswith("../")`?